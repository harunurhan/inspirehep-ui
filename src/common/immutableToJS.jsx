import React from 'react';
import { Iterable } from 'immutable';

export default WrappedComponent => wrappedComponentProps => {
  const propsAsJS = Object.keys(wrappedComponentProps).reduce(
    (newProps, key) => {
      const value = wrappedComponentProps[key];
      // eslint-disable-next-line no-param-reassign
      newProps[key] = Iterable.isIterable(value) ? value.toJS() : value;
      return newProps;
    },
    {}
  );

  return <WrappedComponent {...propsAsJS} />;
};
