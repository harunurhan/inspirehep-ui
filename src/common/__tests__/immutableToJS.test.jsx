import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import immutableToJS from '../immutableToJS';

function MutableDummy(props) {
  return <span {...props} />;
}

const ImmutableDummy = immutableToJS(MutableDummy);

describe('immutableToJS', () => {
  it('converts immutable props to built in js', () => {
    const immutableProp = fromJS({
      list: [{ foo: 'bar1' }, { foo: 'bar2' }],
    });
    const primitiveProp = 'string';
    const mutableProp = {
      array: [{ foo: 'bar' }],
    };
    const wrapper = shallow(
      <ImmutableDummy
        immutableProp={immutableProp}
        mutableProp={mutableProp}
        primitiveProp={primitiveProp}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
