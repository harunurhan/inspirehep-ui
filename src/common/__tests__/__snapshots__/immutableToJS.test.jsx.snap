// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`immutableToJS converts immutable props to built in js 1`] = `
<MutableDummy
  immutableProp={
    Object {
      "list": Array [
        Object {
          "foo": "bar1",
        },
        Object {
          "foo": "bar2",
        },
      ],
    }
  }
  mutableProp={
    Object {
      "array": Array [
        Object {
          "foo": "bar",
        },
      ],
    }
  }
  primitiveProp="string"
/>
`;
